swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -38,89 +38,87 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
+        .providerConfiguration as? [String: String],
+      let logFilter = providerConfiguration[VPNProfileManagerKeys.logFilter]
+    else {
+      completionHandler(
+        PacketTunnelProviderError.savedProtocolConfigurationIsInvalid(
+          "providerConfiguration.logFilter"))
+      return
+    }
 
-      // Hydrate telemetry account slug
-      guard let accountSlug = providerConfiguration[VPNProfileManagerKeys.accountSlug]
-      else {
-        // This can happen if the user deletes the VPN profile while it's
-        // connected. The system will try to restart us with a fresh config
-        // once the user fixes the problem, but we'd rather not connect
-        // without a slug.
-        completionHandler(
-          PacketTunnelProviderError.savedProtocolConfigurationIsInvalid(
-            "providerConfiguration.accountSlug"
-          )
+    // Hydrate telemetry account slug
+    guard let accountSlug = providerConfiguration[VPNProfileManagerKeys.accountSlug]
+    else {
+      // This can happen if the user deletes the VPN profile while it's
+      // connected. The system will try to restart us with a fresh config
+      // once the user fixes the problem, but we'd rather not connect
+      // without a slug.
+      completionHandler(
+        PacketTunnelProviderError.savedProtocolConfigurationIsInvalid(
+          "providerConfiguration.accountSlug"
         )
-        return
-      }
-
-      Telemetry.accountSlug = accountSlug
-
-      let internetResourceEnabled: Bool = if let internetResourceEnabledJSON = providerConfiguration[VPNProfileManagerKeys.internetResourceEnabled]?.data(using: .utf8) {
-        (try? JSONDecoder().decode(Bool.self, from: internetResourceEnabledJSON )) ?? false
-      } else {
-        false
-      }
-
-      let adapter = Adapter(
-        apiURL: apiURL,
-        token: token,
-        id: id,
-        logFilter: logFilter,
-        internetResourceEnabled: internetResourceEnabled,
-        packetTunnelProvider: self
       )
-      self.adapter = adapter
+      return
+    }
 
-      do { try await adapter.start() }
-      catch {
-        Log.error(error)
-        completionHandler(error)
+    Telemetry.accountSlug = accountSlug
 
-        return
-      }
+    let internetResourceEnabled: Bool = if let internetResourceEnabledJSON = providerConfiguration[VPNProfileManagerKeys.internetResourceEnabled]?.data(using: .utf8) {
+      (try? JSONDecoder().decode(Bool.self, from: internetResourceEnabledJSON )) ?? false
+    } else {
+      false
+    }
 
-      // Tell the system the tunnel is up, moving the tunnel manager status to
-      // `connected`.
-      completionHandler(nil)
+    let adapter = Adapter(
+      apiURL: apiURL,
+      token: token,
+      id: id,
+      logFilter: logFilter,
+      internetResourceEnabled: internetResourceEnabled,
+      packetTunnelProvider: self
+    )
+    self.adapter = adapter
```

Comment:
Would it be better to first try to start the adapter before setting it on `self` and leave it uninitialised if it failed?